lazy-pages/src/common.rs

```diff
@@ -222,10 +224,11 @@ impl LazyPagesExecutionContext {
 
     pub fn set_write_accessed(&mut self, page: GearPage) -> Result<(), Error> {
         self.set_accessed(page);
-        match self.write_accessed_pages.insert(page) {
-            true => Ok(()),
-            false => Err(Error::DoubleWriteAccess(page)),
+        if self.write_accessed_pages.contains(page) {
```

Comment:
prev impl was a bit more faster

Reply:
add todo in order to append some return to `insert` and `remove` in `IntervalsTree`

Reply:
If it won't bother you, please ref it in comment there
#3883

Reply:
I have already conntect it to other issue #3875